### ISSUE TITLE
Incorrect timezone and toSeconds interaction

### ISSUE THREAD

**Issue by FedericoCeratto** [nim-lang/Nim#5706]
`echo "TZ ", getTimezone(), " date ", fromSeconds(0)`, when run with a TZ env variable set produces:

TZ=Europe/Riga     -> TZ 0 date 1970-01-01T03:00:00+02:00
TZ=Europe/Dublin   -> TZ 0 date 1970-01-01T01:00:00+00:00

The dates are off by one hour and the timezone is not being detected.

**Comment by treeform** [comment]
I wrote a time/timezone library: https://github.com/treeform/chrono
It returns what I think you want? 
```
    check tsToIso(Timestamp(0.0), tzName = "Europe/Riga") == "1970-01-01T03:00:00+03:00" 
    check tsToIso(Timestamp(0.0), tzName = "Europe/Dublin") == "1970-01-01T01:00:00+01:00" 
```
Feature requests welcome.

**Comment by GULPF** [comment]
This is fixed by #6552, but the result will still be confusing. `getTimezone()` returns the _current_ UTC offset _without_ DST accounted for. But the formatted timezone offset will use the UTC offset that was active at `1970-01-01` UTC _with_ DST accounted for. For those reasons, `getTimezone()` has been deprecated and should not be used.

Here is the current output (which will depend on if the code is executed during DST or not):
```
TZ=Europe/Riga:   TZ -7200 date 1970-01-01T03:00:00+03:00
TZ=Europe/Dublin: TZ 0     date 1970-01-01T01:00:00+01:00
```

**Comment by GULPF** [comment]
This can be closed.
